E0368 needs to be updated to new format

From: src/test/compile-fail/issue-5239-1.rs

Error E0368 needs a span_label, updating it from:

```
error[E0368]: binary assignment operation `+=` cannot be applied to type `&isize`
  --> src/test/compile-fail/issue-5239-1.rs:14:39
   |
14 |     let x = |ref x: isize| -> isize { x += 1; };
   |                                       ^
```

To:

```
error[E0368]: binary assignment operation `+=` cannot be applied to type `&isize`
  --> src/test/compile-fail/issue-5239-1.rs:14:39
   |
14 |     let x = |ref x: isize| -> isize { x += 1; };
   |                                       ^ cannot use `+=` on `isize`
```

--- a/src/librustc_typeck/check/op.rs
+++ b/src/librustc_typeck/check/op.rs
@@ -176,11 +176,15 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 // error types are considered "builtin"
                 if !lhs_ty.references_error() {
                     if let IsAssign::Yes = is_assign {
-                        span_err!(self.tcx.sess, lhs_expr.span, E0368,
-                                  "binary assignment operation `{}=` \
-                                   cannot be applied to type `{}`",
-                                  op.node.as_str(),
-                                  lhs_ty);
+                        struct_span_err!(self.tcx.sess, lhs_expr.span, E0368,
+                                         "binary assignment operation `{}=` \
+                                          cannot be applied to type `{}`",
+                                         op.node.as_str(),
+                                         lhs_ty)
+                            .span_label(lhs_expr.span,
+                                        &format!("cannot use `{}=` on type `{}`",
+                                        op.node.as_str(), lhs_ty))
+                            .emit();
                     } else {
                         let mut err = struct_span_err!(self.tcx.sess, lhs_expr.span, E0369,
                             "binary operation `{}` cannot be applied to type `{}`",

--- a/src/test/compile-fail/issue-5239-1.rs
+++ b/src/test/compile-fail/issue-5239-1.rs
@@ -13,4 +13,5 @@
 fn main() {
     let x = |ref x: isize| -> isize { x += 1; };
     //~^ ERROR E0368
+    //~| NOTE cannot use `+=` on type `&isize`
 }

